crates/sui-adapter/src/adapter.rs

```diff
@@ -509,6 +525,25 @@ pub fn verify_and_link<
     Ok(vm)
 }
 
+/// Given a Vec of `CompiledModule`, runs each module with both the Move VM verifier
+/// and the Sui verifier.
+pub fn run_move_and_sui_bytecode_verifier(
+    modules: &[CompiledModule],
+) -> Result<(), ExecutionError> {
+    // run the Move verifier
+    for module in modules.iter() {
+        verify_module_with_config(Lazy::force(&VERIFIER_CONFIG), module).map_err(|e| {
+            ExecutionError::new_with_source(ExecutionErrorKind::SuiMoveVerificationError, e)
+        })?;
```

Comment:
I think this should just be a `convert_vm_error`

Reply:
The problem is `convert_vm_error ` requires you to pass in `vm: &'r MoveVM`, `state_view: &'r S`, so i figured it was easier to do without.